### ISSUE TITLE
Fix Parquet tests for Databricks 14.3

### ISSUE THREAD

**Issue by razajafri** [NVIDIA/spark-rapids#11531]
Build the plugin against the Databricks 14.3 cluster using #11467. Once built successfully run the parquet tests by `TESTS=parquet_test.py jenkins/databricks/test.sh` 

The following tests fail 
```
[gw2] [ 90%] FAILED ../../src/main/python/parquet_test.py::test_parquet_check_schema_compatibility
```

**Comment by razajafri** [comment]
This can be ported from https://github.com/NVIDIA/spark-rapids/pull/11519

**Comment by mythrocks** [comment]
Hey, @gerashegalov.  Would it be alright if I took this one?

**Comment by gerashegalov** [comment]
OK assigning it to you @mythrocks

**Comment by mythrocks** [comment]
Closing this;  we've addressed the failure in #11519.
